modules/samples/watertank/component/plugin/src/main/java/org/homeautomation/watertank/plugin/impl/DeviceTypeManager.java

```diff
@@ -0,0 +1,257 @@
+                log.warn(msg, iotDAOEx);
+            }
+            String msg = "Error while removing the watertank device : " + deviceId.getId();
+            log.error(msg, e);
+            throw new DeviceManagementException(msg, e);
+        }
+        return status;
+    }
+
+    @Override
+    public boolean isEnrolled(DeviceIdentifier deviceId) throws DeviceManagementException {
+        boolean isEnrolled = false;
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Checking the enrollment of watertank device : " + deviceId.getId());
+            }
+            Device iotDevice =
+                    deviceTypeDAO.getDeviceTypeDAO().getDevice(deviceId.getId());
+            if (iotDevice != null) {
+                isEnrolled = true;
+            }
+        } catch (DeviceMgtPluginException e) {
+            String msg = "Error while checking the enrollment status of watertank device : " +
+                    deviceId.getId();
+            log.error(msg, e);
+            throw new DeviceManagementException(msg, e);
+        }
+        return isEnrolled;
+    }
+
+    @Override
+    public boolean isActive(DeviceIdentifier deviceId) throws DeviceManagementException {
+        return true;
+    }
+
+    @Override
+    public boolean setActive(DeviceIdentifier deviceId, boolean status)
+            throws DeviceManagementException {
+        return true;
+    }
+
+    @Override
+    public Device getDevice(DeviceIdentifier deviceId) throws DeviceManagementException {
+        Device device;
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Getting the details of watertank device : " + deviceId.getId());
+            }
+            device = deviceTypeDAO.getDeviceTypeDAO().getDevice(deviceId.getId());
+        } catch (DeviceMgtPluginException e) {
+            String msg = "Error while fetching the watertank device : " + deviceId.getId();
+            log.error(msg, e);
+            throw new DeviceManagementException(msg, e);
+        }
+        return device;
+    }
+
+    @Override
+    public boolean setOwnership(DeviceIdentifier deviceId, String ownershipType)
+            throws DeviceManagementException {
+        return true;
+    }
+
+    public boolean isClaimable(DeviceIdentifier deviceIdentifier) throws DeviceManagementException {
+        return false;
+    }
+
+    @Override
+    public boolean setStatus(DeviceIdentifier deviceId, String currentOwner,
+                             EnrolmentInfo.Status status) throws DeviceManagementException {
+        return false;
+    }
+
+    @Override
+    public License getLicense(String s) throws LicenseManagementException {
+        return null;
+    }
+
+    @Override
+    public void addLicense(License license) throws LicenseManagementException {
+
+    }
+
+    @Override
+    public boolean requireDeviceAuthorization() {
+        return true;
```

Comment:
IoT device types are expecting device authorization, so in here it is returning true.